modules/gmc-agent/src/main/java/org/gridgain/service/tracing/GmcSpanExporter.java

```diff
@@ -32,77 +30,60 @@
 import io.opencensus.trace.Status;
 import io.opencensus.trace.Tracing;
 import io.opencensus.trace.export.SpanData;
-import io.opencensus.trace.export.SpanExporter;
 import org.apache.ignite.IgniteLogger;
+import org.apache.ignite.IgniteMessaging;
 import org.apache.ignite.internal.GridKernalContext;
 import org.apache.ignite.opencensus.spi.tracing.OpenCensusTraceExporter;
-import org.gridgain.agent.WebSocketManager;
-import org.gridgain.dto.Span;
+import org.apache.ignite.spi.IgniteSpiException;
+import org.gridgain.dto.span.Span;
+import org.gridgain.dto.span.SpanList;
 
 import static java.util.concurrent.TimeUnit.NANOSECONDS;
 import static java.util.concurrent.TimeUnit.SECONDS;
-import static org.gridgain.agent.StompDestinationsUtils.buildSaveSpanDest;
 
 /**
- * Tracing service.
+ * Span exporter which send spans to coordinator.
  */
-public class TracingService implements AutoCloseable {
+public class GmcSpanExporter implements AutoCloseable {
+    /** Status description. */
+    public static final String TRACING_TOPIC = "gmc-tracing-topic";
+
     /** Status code. */
     private static final String STATUS_CODE = "census.status_code";
 
     /** Status description. */
     private static final String STATUS_DESCRIPTION = "census.status_description";
 
-    /** Status description. */
-    private static final String HANDLER_NAME = "gmc";
-
     /** Context. */
     private GridKernalContext ctx;
 
-    /** Manager. */
-    private WebSocketManager mgr;
-
     /** Logger. */
     private IgniteLogger log;
 
-    /** Handler. */
-    private SpanExporter.Handler hnd;
-
     /** Exporter. */
     private OpenCensusTraceExporter exporter;
 
     /**
      * @param ctx Context.
-     * @param mgr Manager.
      */
-    public TracingService(GridKernalContext ctx, WebSocketManager mgr) {
+    public GmcSpanExporter(GridKernalContext ctx) {
         this.ctx = ctx;
-        this.mgr = mgr;
-        this.log = ctx.log(TracingService.class);
-
-        hnd = getTraceHandler();
-    }
+        this.log = ctx.log(GmcSpanExporter.class);
 
-    /**
-     * Register span exporter handler.
-     */
-    public void registerHandler() {
         if (ctx.config().getTracingSpi() != null) {
-            exporter = new OpenCensusTraceExporter(hnd);
-            exporter.start(ctx.igniteInstanceName());
+            try {
+                exporter = new OpenCensusTraceExporter(getTraceHandler());
+                exporter.start(ctx.igniteInstanceName());
+            }
+            catch (IgniteSpiException ex) {
+                log.error("Can't start the trace exporter", ex);
+            }
         }
     }
 
-    /**
-     * Send buffered spans.
-     */
-    public void sendInitialState() {
-        hnd.export(Collections.emptyList());
-    }
-
     /** {@inheritDoc} */
     @Override public void close() {
-        if (ctx.config().getTracingSpi() != null)
+        if (ctx.config().getTracingSpi() != null && exporter != null)
```

Comment:
Do we need to check `ctx.config().getTracingSpi() != null` ? Looks like not needed check.

Reply:
Agree.